### DISCUSSION TITLE
Switch back language at login to DE

### DISCUSSION THREAD

**Discussion by EmEiBee** [farag2/Sophia-Script-for-Windows#198] [Q&A] [answered]
Hello all,

very great tool at all! I try it now more then 5th and it works like a charm... but...
I noticed that the language at the 1st login screen is set back to English. I can chose German (normal and IBM) from the bottom right menu but that sucks a bit.
It also don't help if I set German as default inside WIN. After reboot login are back to English.

I have read the script uncounted up- and down but can't find the line that handle that...

So who knows what I have to do to switch it back zu German, best as a script line I can insert to run it at the end?

Thx you all very much for your assistance ...

Greetz
Micha

**Comment by farag2** [comment]
https://github.com/farag2/Windows-10-Sophia-Script/blob/40dfbce6c6fd0d757b75dd4c4905a7f75d190af2/Sophia/PowerShell%205.1/Module/Sophia.psm1#L5439

**Reply by farag2** [reply]
:D

**Comment by farag2** [comment]
https://github.com/farag2/Windows-10-Sophia-Script/blob/40dfbce6c6fd0d757b75dd4c4905a7f75d190af2/Sophia/PowerShell%205.1/Sophia.ps1#L729

**Reply by EmEiBee** [reply]
... very fast reply!!! Ty a lot!

But unfortunately I'm not a coder and I don't really know what to do with it ... If I understand correctly, is that the line that the system forces to English ?!
So the questions are:

1. Do I have to delete the line for future use?

2. How do I get my system, which has already been set in English, back into German?

Greetz
Micha

EDIT say...

```powershell
function SetInputMethod
{
	param
	(
		[Parameter(
			Mandatory = $true,
			ParameterSetName = "Deutsch"
		)]
		[switch]
		$Deutsch,

		[Parameter(
			Mandatory = $true,
			ParameterSetName = "Default"
		)]
		[switch]
		$Default
	)

	switch ($PSCmdlet.ParameterSetName)
	{
		"Deutsch"
		{
			Set-WinDefaultInputMethodOverride -InputTip "0407:00000407"
		}
		"Default"
		{
			Remove-ItemProperty -Path "HKCU:\Control Panel\International\User Profile" -Name InputMethodOverride -Force -ErrorAction SilentlyContinue
		}
	}
}
```

Seems to be that here all is set to German. After Login it's ok and German is the default but while login that's not. DO I misunderstand some?

**Reply by farag2** [reply]
1. No. All stuff does the preset. As you configure it, that way it will work. Open Sophia.ps1 and find `SetInputMethod -English`. Comment it our by placing the `#` char in the beginning of the string. Find a few strings lower `SetInputMethod -Default` and remove the `#` char from the beginning.

https://github.com/farag2/Windows-10-Sophia-Script/blob/40dfbce6c6fd0d757b75dd4c4905a7f75d190af2/Sophia/PowerShell%205.1/Sophia.ps1#L729

2. This action reverts the changes you made to the default value.

You may even try to invoke the specific function `SetInputMethod -Default`, if you read the [instruction](https://github.com/farag2/Windows-10-Sophia-Script#how-to-run-the-specific-functions)
It will look like `Sophia -Functions "SetInputMethod -Default"`

**Comment by farag2** [comment, marked as answer]
If you want to highlight the syntax use the following construction

````
```powershell
code is here
```
````

**Reply by EmEiBee** [reply]
...AHHHH! Ok, now I got it ;)
(some minutes are gone .... tick tock tick tock)... No, that don't do it...

I have used the GUI for it and create and save the script as "debloat.ps1". And there is set "SetInputMethod -Default". But as you can see the login is set to English again ...
![sophia_01](https://user-images.githubusercontent.com/30090331/122351357-58d9ce80-cf4e-11eb-911d-7d8129289a9b.jpg)

PS: I use Notepad++ ...

**Comment by farag2** [comment]
Did you run this?

```powershell
Remove-ItemProperty -Path "HKCU:\Control Panel\International\User Profile" -Name InputMethodOverride -Force -ErrorAction SilentlyContinue
```

> I have used the GUI for it

Why did you use the wrapper for 1 function.

**Reply by EmEiBee** [reply]
>> Why did you use the wrapper for 1 function. <<
... no, that is an misunderstanding ... I have used the GUI to create a script as my preferred default script. The Notebook from I have take the screenshot is an experimental one I can switch back to an fresh installed WIN how often I need to check out if all working well (booting from an fresh image copy).

I now try to run the line you gave me... (time goes by ...) 
I have run that but that change anything... Startup- logon are still English...

_May can I plain change `SetInputMethod -Default` to `SetInputMethod -Deutsch` to force?_
Quack... Knot in brain... Forget it ;) That can't work ...

**Reply by EmEiBee** [reply]
... Ok, now I got it!
I have try `Set-WinDefaultInputMethodOverride -InputTip "0407:00000407"` in an administrative PS and now it works as expected!

Thank you very much for your assistance! You do a very great and nice work at all!

Greetz
Micha

**Comment by farag2** [comment]
<https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/default-input-locales-for-windows-language-packs>

You're god damn right! I got confused what did you want. Yeah, to set the German language as the first lang on logon screen, you've chosen the right lang ID. 😸

**Reply by EmEiBee** [reply]
... sometimes even a blind chicken (I) find a grain ;)
